airflow/providers/google/cloud/triggers/cloud_run.py

```diff
@@ -102,21 +102,21 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
         while timeout is None or timeout > 0:
             operation: operations_pb2.Operation = await hook.get_operation(self.operation_name)
             if operation.done:
-                # An operation can only have one of those two combinations: if it is succeeded, then
-                # the response field will be populated, else, then the error field will be.
-                if operation.response is not None:
+                # An operation can only have one of those two combinations: if it is failed, then
+                # the error field will be populated, else, then the response field will be.
+                if operation.error.SerializeToString():
                     yield TriggerEvent(
                         {
-                            "status": RunJobStatus.SUCCESS,
+                            "status": RunJobStatus.FAIL.value,
+                            "operation_error_code": operation.error.code,
+                            "operation_error_message": operation.error.message,
                             "job_name": self.job_name,
                         }
                     )
                 else:
                     yield TriggerEvent(
                         {
-                            "status": RunJobStatus.FAIL,
-                            "operation_error_code": operation.error.code,
-                            "operation_error_message": operation.error.message,
+                            "status": RunJobStatus.SUCCESS.value,
                             "job_name": self.job_name,
                         }
                     )
```

Comment:
According to my tests, this field still exists even if there is no actual error. Only after converting this field to String we can get b'' empty string and see that there is no actual error. 
The same story is with response filed. It is always included to operation but without actual content.   
That was the origin of this issue, the response field always contained something, even in case of error, and since it was the first check after getting Operation, every time we got Succeeded status even if actually the job has failed

Reply:
Perhaps, it makes sense to mention it in description in order to pass community code review faster.